### PR TITLE
Add User, UserRepository Classes and Associated Tests

### PR DESCRIPTION
**Pull request by ttarlov**
# PR Description
Please include a summary of the change:

- Adds UserRepository Class with methods: calculateAvgStepGoal, getUserDataById. 
- Adds User Class with method: getFirstName. 
- Adds Mocha/Chai tests for User and UserRepository Classes.

:crystal_ball: JS modifications/updates.

If it is a new feature, describe the feature and the iteration of the project spec it addresses.

Note any questions or struggles as well.

## Type of change
Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code follows the Turing style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: test/user-repository-test.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    });
+  });
+});
+
+
+
+
+
+//
```
Comment (Karlfunhouse) [inline test/user-repository-test.js]: can delete this extra //

File: test/user-test.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    expect(singleUser).to.be.an.instanceof(User);
+  });
+
+  describe('getFirstName Method', () => {
+
+    it('should return first name', () => {
+    expect(singleUser.getFirstName()).to.equal('Luisa');
+    });
+  });
+
```
Comment (Karlfunhouse) [inline test/user-test.js]: delete extra spacing

File: src/userrepository.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    //need to loop through all of the users. grab their dailyStepGoal
+    //add them together and divide them by the number of users in the group.
+    let total = 0;
+    this.data.forEach(user => {
+      return total+= user.dailyStepGoal;
+    });
+    let avg = total / this.data.length;
+    return avg;
+  }
+
```
Comment (Karlfunhouse) [inline src/userrepository.js]: delete white space

Comment (ttarlov) [reply src/userrepository.js]: oh yeah. good catch.

**Pull request merged**